.buildkite/scripts/steps/artifacts/docker_image.sh

```diff
@@ -126,15 +126,14 @@ steps:
       env:
         CONTAINER: "$KIBANA_IMAGE"
     soft_fail: true
-  - label: ":argo: Update kibana image tag for kibana-controller using gpctl"
+  - label: ":serverless::argo: Run synthetics tests and update kibana image tag to ${GIT_ABBREV_COMMIT} for kibana-controller"
     branches: main
-    trigger: gpctl-promote-with-e2e-tests
+    trigger: gpctl-promote-after-serverless-devenv-synthetics
     build:
       env:
         SERVICE_COMMIT_HASH: "$GIT_ABBREV_COMMIT"
         SERVICE: kibana
         REMOTE_SERVICE_CONFIG: https://raw.githubusercontent.com/elastic/serverless-gitops/main/gen/gpctl/kibana/dev.yaml
-        DRY_RUN: "${DRY_RUN:-false}"
```

Comment:
Is this no longer available?  It was previously helpful for us to run this pipeline without promoting artifacts in support of testing pipeline changes on our side.

Reply:
This functionality was also not available in the [previous pipeline](https://github.com/elastic/gpctl/blob/main/.buildkite/pipeline.promote-ci-with-e2e-tests.yml). 

Out of curiosity, have you tried to call this recently with DRY_RUN=true?
We do have a separate pipeline that only does the testing and no promotion.

Reply:
I'm mostly not involved with the Kibana release pipeline, not sure.

@pheyos @delanni can you take a look?

Reply:
We used this flag while testing the migrations to ensure we triggered the right downstream. I don't know if it's still in use, or would have future uses.

For more context, we added the Kibana triggers here: https://github.com/elastic/kibana/pull/178658/files

the corresponding escape hatch was added to `promote.sh` here: https://github.com/elastic/gpctl/pull/261

It's true, that the `gpctl-promote-with-e2e-tests` pipeline didn't have a reference to `DRY_RUN`, but I believe, it did work for our case using the `gpctl-promote` pipeline, as we directly triggered that (and thus the env section from the first linked PR was passed on). Through `gpctl-promote-with-e2e-tests` this is probably lost, unless we pass along the env to the trigger step. 

I don't think we're using the `DRY_RUN` for any other use cases right now. So we can either add env passing there, or allow this to be removed for now. I'd go for the first option, since the hatch exists in `promote.sh`, we may as well expose it.

Reply:
I've added support for a more explicit env variable called `GPCTL_PROMOTE_DRY_RUN` in https://github.com/elastic/serverless-gitops/pull/4202.